admin/sql/CreateFunctions.sql

```diff
@@ -1255,6 +1255,86 @@ BEGIN
 END;
 $$ LANGUAGE 'plpgsql';
 
+-------------------------------------------------------------------
+-- Ratings
+-------------------------------------------------------------------
+
+CREATE OR REPLACE FUNCTION update_aggregate_rating(entity_type ratable_entity_type, entity_id INTEGER)
+RETURNS VOID AS $$
+BEGIN
+  -- update the aggregate rating for the given entity_id.
+  EXECUTE format(
+    $SQL$
+      UPDATE %2$I
+         SET rating = agg.rating,
+             rating_count = nullif(agg.rating_count, 0)
+        FROM (
+          SELECT count(rating)::INTEGER AS rating_count,
+                 -- trunc(x + 0.5) is used because round() on REAL values
+                 -- rounds to the nearest even number.
```

Comment:
So the idea is that it always truncates down, so by adding 0.5 it will round down to the *next* int if it was .5 or higher? I guess it makes sense.

Reply:
Yeah.  This is basically copied from what the Perl was previously doing (`int($rating_sum / $rating_count + 0.5)`).  (Though in that case I think we could've just used `POSIX::round`).